mindmeld/text_preparation/tokenizers.py

```diff
@@ -0,0 +1,199 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2015 Cisco Systems, Inc. and others.  All rights reserved.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This module contains Tokenizers."""
+
+from abc import ABC, abstractmethod
+import logging
+import unicodedata
+
+from .spacy_model_factory import SpacyModelFactory
+
+logger = logging.getLogger(__name__)
+
+
+class Tokenizer(ABC):
+    """Abstract Tokenizer Base Class."""
+
+    def __init__(self):
+        """Creates a tokenizer instance."""
+        pass
+
+    @abstractmethod
+    def tokenize(self, text):
+        """
+        Args:
+            text (str): Input text.
+        Returns:
+            tokens (List[str]): List of tokens.
+        """
+        raise NotImplementedError("Subclasses must implement this method")
+
+
+class NoOpTokenizer(Tokenizer):
+    """A No-Ops tokenizer."""
+
+    def __init__(self):
+        """Initialize the NoOpTokenizer."""
+        pass
+
+    def tokenize(self, text):
+        """
+        Args:
+            text (str): Input text.
+        Returns:
+            tokens (List[str]): List of tokens.
+        """
+        return [text]
+
+
+class CharacterTokenizer(Tokenizer):
+    """A Tokenizer that splits text at the character level."""
+
+    def __init__(self):
+        """Initializes the CharacterTokenizer."""
```

Comment:
Looking at what other tools are calling their tokenizers, I think this should be called a LetterTokenizer, comparable to (though different than) [Elasticsearch](https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-letter-tokenizer.html) and [AllenNLP](https://docs.allennlp.org/main/api/data/tokenizers/letters_digits_tokenizer/). As you suggested, a CharacterTokenizer (which may also be useful) has single characters in each token.

Reply:
Thanks for these resources! Sure, I'll go ahead and rename this to a LetterTokenizer and also create a separate CharacterTokenizer

Reply:
Added a separate CharacterTokenizer